source/gui/keybinding_manager.py

```
@@ -0,0 +1,372 @@
+        self.log("[KEYBIND] Executing: Set wallpaper")
+        
+        if not self.config.get("--dir"):
+            self.log("[KEYBIND ACTION] No directory selected")
+            messagebox.showwarning(
+                "No Directory",
+                "Please select a wallpaper directory first"
+            )
+            return
+        
+        # Delegate to event handler which has the UI logic
+        from tkinter import filedialog
```

Ruff (0.14.14)

[error] Redefinition of unused `filedialog` from line 15: `filedialog` redefined here

Remove definition: `filedialog`

(F811)